src/libinjector/injector_stack.c

```diff
@@ -234,6 +242,142 @@ static addr_t place_struct_on_stack_64(vmi_instance_t vmi, x86_registers_t* regs
     return status == VMI_FAILURE ? 0 : addr;
 }
 
+static addr_t place_argument_on_addr_32(vmi_instance_t vmi, x86_registers_t* regs, struct argument* arg, addr_t addr)
+{
+    switch (arg->type)
+    {
+        case ARGUMENT_STRING:
+        {
+            addr = place_string_on_stack_32(vmi, regs, addr, arg->data, arg->size);
+            if ( !addr ) goto err;
+            arg->data_on_stack = addr;
+            break;
+        }
+        case ARGUMENT_STRUCT:
+        {
+            addr = place_struct_on_stack_32(vmi, regs, addr, arg->data, arg->size);
+            if ( !addr ) goto err;
+            arg->data_on_stack = addr;
+            break;
+        }
+        case ARGUMENT_INT:
+        {
+            arg->data_on_stack = (uint64_t)arg->data;
+            addr -= 0x4;
+
+            ACCESS_CONTEXT(ctx,
+                .translate_mechanism = VMI_TM_PROCESS_DTB,
+                .dtb = regs->cr3,
+                .addr = addr
+            );
+
+            if (VMI_FAILURE == vmi_write_32(vmi, &ctx, (uint32_t*)&arg->data_on_stack))
+            {
+                fprintf(stderr, "vmi_write_32 failed\n");
```

Comment:
`PRINT_DEBUG` and a bit more meaningful message please

Reply:
`PRINT_DEBUG` is defined in `private.h` so i wasn't sure if I should have included that file here. Do I move the macro to some different file ( `libinjector.h` perhaps? ) or should I include `private.h` file in this?

Reply:
Include private.h here

Reply:
Please have a look at https://github.com/tklengyel/drakvuf/pull/1305. I have done the concerning changes there